extensions/sql-migration/src/dialog/skuRecommendationResults/SelectStorageAccountDialog.ts

```diff
@@ -589,28 +589,49 @@ export class SelectStorageAccountDialog {
 		const containerName = this._blobContainer.name;
 		const templates = this.migrationStateModel._armTemplateResult.templates!;
 		const sharedKeyCredential = new StorageSharedKeyCredential(this._storageAccount.name, storageKeys.keyName1);
-
-		const sasToken = generateBlobSASQueryParameters({
-			containerName,
-			permissions: BlobSASPermissions.parse("racwd"),
-			expiresOn: new Date(new Date().valueOf() + 86400),
-		},
+		var deployToAzureUrl: string = "";
+
+		const sasOptions = {
+			services: AccountSASServices.parse("b").toString(),          // blobs
+			resourceTypes: AccountSASResourceTypes.parse("sco").toString(), // service, container, object
+			permissions: AccountSASPermissions.parse("rwdlacupi"),          // permissions
+			protocol: SASProtocol.Https,
+			startsOn: new Date(),
+			expiresOn: new Date(new Date().valueOf() + (30 * 60 * 1000)),   // 30 minutes
+		};
+
+		const sasToken = generateAccountSASQueryParameters(
+			sasOptions,
 			sharedKeyCredential
 		).toString();
 
 		try {
 			for (let i = 0; i < templates.length; i++) {
 				const blobName = utils.generateTemplatePath(this.migrationStateModel, this._targetType, i + 1);
-				const sasUrl = `https://${accountName}.blob.core.windows.net/${containerName}/${blobName}?${sasToken}`;
+				var sasUrl = `https://${accountName}.blob.core.windows.net/${containerName}/${blobName}?${sasToken}`;
+				deployToAzureUrl = sasUrl;
 				const blockBlobClient = new BlockBlobClient(sasUrl);
 				await blockBlobClient.upload(templates[i], templates[i].length);
 			}
 			void vscode.window.showInformationMessage(constants.UPLOAD_TEMPLATE_SUCCESS);
+			this.DeployToAzure(deployToAzureUrl);
 		}
 		catch (e) {
 			logError(TelemetryViews.UploadArmTemplateDialog, 'ArmTemplateUploadError', e);
 			void vscode.window.showErrorMessage(constants.UPLOAD_TEMPLATE_FAIL);
 		}
 
 	}
+
+	private DeployToAzure(deployToAzureUrl: string) {
+		deployToAzureUrl = encodeURIComponent(deployToAzureUrl);
```

Comment:
Lets have a separate variable instead of reusing the one that is sent as part of the method.

Reply:
resolving since the code has been changed.